processing/src/main/java/org/apache/druid/segment/DefaultColumnFormatConfig.java

```diff
@@ -21,12 +21,24 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.error.DruidException;
 
 import javax.annotation.Nullable;
 import java.util.Objects;
 
 public class DefaultColumnFormatConfig
 {
+  public static void validateNestedFormatVersion(@Nullable Integer formatVersion)
+  {
+    if (formatVersion != null) {
+      if (formatVersion < 4 || formatVersion > 5) {
+        throw DruidException.forPersona(DruidException.Persona.USER)
+                            .ofCategory(DruidException.Category.INVALID_INPUT)
+                            .build("Unsupported nested column format version[%s]", formatVersion);
```

Comment:
Nit - can you add in the error message that supported values are 4 and 5?

Reply:
i think im going to skip this for now to not churn through CI again, and also seems like we should find some programmatic way to list what versions are supported so the error messaging doesn't become stale if we add another version. Also the versions are still kind of confusing so idk how I feel about exposing them too much, 4 makes sense, but 5 is a virtual version since 5 is really version 0 of the new nested common format introduced by 'auto'.